driver/internal/protocol/authentication_jwt.go

```diff
@@ -0,0 +1,226 @@
+package protocol
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/SAP/go-hdb/driver/internal/protocol/encoding"
+	"github.com/SAP/go-hdb/driver/unicode/cesu8"
+)
+
+const mnJWT = "JWT"
+
+func checkMethodJWT(method string) error {
+	if !strings.EqualFold(mnJWT, method) {
+		return fmt.Errorf("unexpected method %s - expected %s", method, mnJWT)
+	}
+
+	return nil
+}
+
+type authMethodJWT struct {
+	token string
+}
+
+func (m *authMethodJWT) String() string {
+	return fmt.Sprintf("method %s token %s", mnJWT, m.token)
+}
+
+func (m *authMethodJWT) size() int {
+	size := 1 // len byte for method name
+	size += len(mnJWT)
+
+	size += varBytesSize(len(m.token)) // token can be longer than bytesLenIndSmall bytes
+
+	return size
+}
+
+func (m *authMethodJWT) decode(dec *encoding.Decoder, ph *partHeader) error {
+	panic("not implemented")
+}
+
+func (m *authMethodJWT) encode(enc *encoding.Encoder) error {
+	if err := authShortBytes.encode(enc, []byte(mnJWT)); err != nil {
+		return err
+	}
+
+	return encodeVarBytes(enc, []byte(m.token))
+}
+
+type authInitReqJWT struct {
+	method *authMethodJWT
+}
+
+func (r *authInitReqJWT) String() string {
+	return fmt.Sprintf("emptyusername method %v", r.method)
+}
+
+func (r *authInitReqJWT) size() int {
+	size := int16Size // no of parameters
+
+	size++ // len byte for empty username
+
+	size += r.method.size()
+
+	return size
+}
+
+func (r *authInitReqJWT) encode(enc *encoding.Encoder) error {
+	enc.Int16(int16(1 + 1*2)) // empty username + one method with two fields (method name + token)
+
+	// empty user name
+	if err := authShortCESU8String.encode(enc, ""); err != nil {
+		return err
+	}
+
+	return r.method.encode(enc)
+}
+
+func (r *authInitReqJWT) decode(dec *encoding.Decoder, ph *partHeader) error {
+	panic("not implemented")
```

Comment:
my understanding is that for requests the decode method is not relevant. Or are there scenarios where `decode` will be invoked?